R/package.R

```diff
@@ -91,27 +82,18 @@ send_headers <- c("Accept" = "application/vnd.github.v3+json",
 #'
 
 gh <- function(endpoint, ..., .token = NULL,
-               .api_url = Sys.getenv('GITHUB_API_URL', unset = default_api_url),
-               .limit = NULL, .send_headers = NULL) {
-
-  if (is.null(.token)) .token <- gh_token()
-
-  params <- list(...)
-
-  parsed <- parse_endpoint(endpoint, params)
-  method <- parsed$method
-  endpoint <- parsed$endpoint
-  params <- parsed$params
+               .api_url = NULL, .limit = NULL, .send_headers = NULL) {
 
-  auth <- get_auth(.token)
-  headers <- get_headers(.send_headers)
+  req <- gh_build_request(endpoint = endpoint, params = list(...),
+                          token = .token, send_headers = .send_headers,
+                          api_url = .api_url)
 
-  url <- URLencode(paste0(.api_url, endpoint))
+  raw <- gh_make_request(req)
 
-  res <- gh_url(method, url, auth, headers, params)
+  res <- gh_process_response(raw)
 
-  while (! is.null(.limit) && length(res) < .limit && gh_has_next(res)) {
-    res2 <- gh_next(res, .token = .token)
+  while (!is.null(.limit) && length(res) < .limit && gh_has_next(res)) {
+    res2 <- gh_next(res)
```

Comment:
The token is already in the response, as part of the headers. So I just re-use the headers for calls for other pages. It doesn't seem like the token needs to be an argument? Or am I missing something?